app/__init__.py

```diff
@@ -0,0 +1,6 @@
+from flask import Flask
+
+def create_app(test_config=None):
```

Comment:
I see the current learn material shows the `test_config` param, but the current learn text around configurations feels like a non sequitur. We don't even use the parameter (poorly, at that) until we get to testing (part 6). Could we leave this off for now and not introduce the idea until we need it?

Reply:
I would love to leave that param off until testing. If other folks are on board, I'll make that update, I would like to chat with folks if there's another way we want to set up that param as well since we don't use the contents we send meaningfully.

Reply:
Yeah. As right now, we basically just use it as a boolean to determine from which environment variables to read the settings. What makes more sense to me is that if we pass in a config dictionary (for whatever reason, it needn't be test-specific), we should merge those values into the environment, and the code should always look in the "correct" key rather than needing to know to change keys depending on whether we are testing. The test startup should just pass the environment overrides it needs.